src/core/ConfigManager.cpp

```diff
@@ -338,6 +349,34 @@ void ConfigManager::addRecentlyOpenedProject(const QString & file)
 
 
 
+void ConfigManager::addFavoriteItem(QString& item)
+{
+	item = QDir::toNativeSeparators(removeTrailingSeparators(item));
+
+	auto instance = inst();
+	instance->m_favoriteItems.push_back(item);
+	instance->saveConfigFile();
+}
```

Comment:
Editing the parameter in this case is unusual. `QString` uses implicit sharing, so I believe this will edit the original string, which isn't necessary and could be unintended. I recommend passing `QDir::toNativeSeparators(removeTrailingSeparators(path))` to the functions instead of the internal modification trickery. This goes for the other functions as well.

```suggestion
void ConfigManager::addFavoriteItem(const QString& path)
{
	auto instance = inst();
	instance->m_favoriteItems.push_back(item);
	instance->saveConfigFile();
}
```